src/Language/DifferentialDatalog/Validate.hs

```diff
@@ -272,18 +278,41 @@ ruleRHSValidate d rl@Rule{..} (RHSFlatMap v e) idx = do
     exprValidate d [] ctx e
     case exprType' d ctx e of
          TOpaque _ tname [_] | elem tname sET_TYPES -> return ()
-         t  -> err (pos e) $ "FlatMap expression must be of these types: " ++ intercalate ", " sET_TYPES ++ ", but its type is " ++ show t
+         TOpaque _ "std.Map" [_,_]                  -> return ()
+         t  -> err (pos e) $ "FlatMap expression must have one of these types: " ++ intercalate ", " sET_TYPES ++ ", or std.Map but its type is " ++ show t
```

Comment:
Does this take care of the case where you typedefed one of these standard types and are using the new name?

Reply:
These are declared in `stdlib.dl` that is automatically imported by all other modules. DDlog will complain if the user attempts to re-define these types.

Reply:
The question is the other way round:
`typedef X Set<string, string>`
And you are using X.

Reply:
Yes, the `exprType'` method will expand type definitions.